src/test/txreconciliation_tests.cpp

```diff
@@ -81,4 +87,193 @@ BOOST_AUTO_TEST_CASE(IsPeerRegisteredTest)
+                                           /*inbounds_fanout_tx_relay=*/0, /*outbounds_fanout_tx_relay=*/0));
+    }
+
+    // Once the peer is registered, it should be selected for flooding of some transactions.
+    // Since there is only one reconciling peer, it will be selected for all transactions.
+    BOOST_REQUIRE_EQUAL(tracker.RegisterPeer(peer_id0, /*is_peer_inbound=*/false, 1, 1), ReconciliationRegisterResult::SUCCESS);
+    for (int i = 0; i < 100; ++i) {
+        BOOST_CHECK(tracker.ShouldFanoutTo(Wtxid::FromUint256(frc.rand256()), peer_id0,
+                                           /*inbounds_fanout_tx_relay=*/0, /*outbounds_fanout_tx_relay=*/0));
+    }
+
+    // Don't select a fanout target if it was already fanouted sufficiently.
+    for (int i = 0; i < 100; ++i) {
+        BOOST_CHECK(!tracker.ShouldFanoutTo(Wtxid::FromUint256(frc.rand256()), peer_id0,
+                                            /*inbounds_fanout_tx_relay=*/0, /*outbounds_fanout_tx_relay=*/1));
+    }
+
+    tracker.ForgetPeer(peer_id0);
+    // A forgotten (reconciliation-wise) peer should be always selected for fanout again.
+    for (int i = 0; i < 100; ++i) {
+        BOOST_CHECK(tracker.ShouldFanoutTo(Wtxid::FromUint256(frc.rand256()), peer_id0,
+                                           /*inbounds_fanout_tx_relay=*/0, /*outbounds_fanout_tx_relay=*/0));
+    }
+
+    // Now for inbound connections.
+    // Initialize a new instance with a new hasher to be used later on.
+    CSipHasher hasher2(0x0706050403020100ULL, 0x4F0E0D0C0B0A0908ULL);
+    TxReconciliationTracker tracker2(TXRECONCILIATION_VERSION, hasher2);
+    int inbound_peers = 36;
+    for (int i = 1; i < inbound_peers; ++i) {
+        tracker.PreRegisterPeer(i);
+        tracker2.PreRegisterPeer(i);
+        BOOST_REQUIRE_EQUAL(tracker.RegisterPeer(i, /*is_peer_inbound=*/true, 1, 1), ReconciliationRegisterResult::SUCCESS);
+        BOOST_REQUIRE_EQUAL(tracker2.RegisterPeer(i, /*is_peer_inbound=*/true, 1, 1), ReconciliationRegisterResult::SUCCESS);
+    }
+
+    // Relay to a fraction of registered inbound peers.
+    // For 35 peers we will choose 3.5 flooding targets, which means that it's either 3 or 4 with
+    // 50% chance. Make sure the randomness actually works by checking against a different hasher.
+    size_t total_fanouted1 = 0;
+    size_t total_fanouted2 = 0;
+    auto wtxid = Wtxid::FromUint256(uint256(1)); // determinism is required.
+    for (int i = 1; i < inbound_peers; ++i) {
+        total_fanouted1 += tracker.ShouldFanoutTo(wtxid, i,
+                                                  /*inbounds_fanout_tx_relay=*/0, /*outbounds_fanout_tx_relay=*/0);
+        total_fanouted2 += tracker2.ShouldFanoutTo(wtxid, i,
+                                                   /*inbounds_fanout_tx_relay=*/0, /*outbounds_fanout_tx_relay=*/0);
+    }
+    BOOST_CHECK_EQUAL(total_fanouted1, 3);
+    BOOST_CHECK_EQUAL(total_fanouted2, 4);
+
```

Comment:
```suggestion
for (int i = 1; i < inbound_peers; ++i) {
        // create 4000 random wtxids so that FANOUT_TARGETS_PER_TX_CACHE_SIZE is breached
        for (int j = 0; j < 4000; ++j) {
            tracker.ShouldFanoutTo(Wtxid::FromUint256(frc.rand256()), i,
                                   /*inbounds_fanout_tx_relay=*/0, /*outbounds_fanout_tx_relay=*/0);
        }
    }
```

increases coverage of the units tests to check the cache_size check in ```IsFanoutTarget``` .

Reply:
I haven't taken this yet, given I'm working on a last commit reworking part of how the fanout selection works. I may consider it after that. In any case, looks like this is only filling the cache but not testing anything after it

Reply:
Resolving since we don't have a cache anymore